### PR TITLE
prevent errors and warnings

### PR DESCRIPTION
**Pull request by conalli**
temporary/permanent fixes to reduce errors and warnings, making future errors easier to track

### CODE REVIEW

File: gui.tscn [hunk 1]
```diff
@@ -261,6 +261,4 @@ offset_bottom = 202.0
 text = "  No  "
 
 [connection signal="pressed" from="GUI/BuildMenu/BuildButton" to="GUI/BuildMenu" method="_on_build_button_pressed"]
-[connection signal="mouse_entered" from="GUI/BuildMenu/RoomPanel/BuildCloseButton" to="GUI/BuildMenu" method="_on_room_panel_mouse_entered"]
```
Comment (conalli) [inline gui.tscn:264]: unused - generated missing method error

File: project.godot [hunk 2]
```diff
@@ -13,7 +13,7 @@ config_version=5
 config/name="Jupiter's Forge"
 run/main_scene="res://scene.tscn"
 config/features=PackedStringArray("4.0", "Forward Plus")
-config/icon="res://icon.svg"
+config/icon="res://assets/sprites/flirty_pose.png"
```
Comment (conalli) [inline project.godot:16]: temporarily changed icon as icon.svg doesn't exist - generated error

File: scripts/scene/BuildingManager.gd [hunk 3]
```diff
@@ -102,6 +102,11 @@ func _on_building_state_input(event: InputEvent) -> void:
 func _on_selecting_roomtype_state_entered() -> void:
 	build_menu.show_room_panel(room_types)
 
+func _on_selecting_roomtype_state_input(event: InputEvent):
```
Comment (conalli) [inline scripts/scene/BuildingManager.gd:105]: apologies for adding unrelated code to the pr but while building i felt we needed to allow right click on selecting_roomtype to allow stop building as it works for all other steps

File: scripts/scene/RoomBuilder.gd [hunk 4]
```diff
@@ -32,13 +32,13 @@ var popup_message: String
...
 enum Action {BACK, FORWARD, COMPLETE}
 
-func _init(gui: GUI, station: Station, base_tile_map: TileMap, build_tile_map: TileMap, rooms: Array[Room], room_types: Array[RoomType]) -> void:
-	self.gui = gui
-	self.station = station
-	self.base_tile_map = base_tile_map
-	self.build_tile_map = build_tile_map
-	self.rooms = rooms
-	self.room_types = room_types
+func _init(gui_node: GUI, station_node: Station, base_tile_map_node: TileMap, build_tile_map_node: TileMap, rooms_arr: Array[Room], room_types_arr: Array[RoomType]) -> void:
```
Comment (conalli) [inline scripts/scene/RoomBuilder.gd:35]: ugly naming i know but was generating a shadowing error

File: scripts/scene/RoomBuilder.gd [hunk 5]
```diff
@@ -50,10 +50,10 @@ func stop_drafting() -> void:
 
 # --- Input functions ---
 
-func selecting_tile(event: InputEventMouseButton, offset: Vector2, zoom: Vector2, selected_room_type: RoomType) -> void:
+func selecting_tile(event: InputEventMouseButton, offset: Vector2, zoom: Vector2, current_room_type: RoomType) -> void:
```
Comment (conalli) [inline scripts/scene/RoomBuilder.gd:53]: same shadowing error as above

File: scripts/scene/RoomBuilder.gd [hunk 6]
```diff
@@ -225,7 +225,7 @@ func generate_unique_room_id() -> int:
 	return unique_id
 
 func check_room_id_exists(room_id: int) -> bool:
-	return room_id in rooms
+	return rooms.any(func(room: Room): return room.id == room_id)
```
Comment (conalli) [inline scripts/scene/RoomBuilder.gd:228]: previous function was generating errors

File: scripts/ui/GUI.gd [hunk 7]
```diff
@@ -46,7 +46,7 @@ func update_clock(in_game_time) -> void:
 	var minutes = int((in_game_time % 3600) / 60)
 	$Clock/Time.text = str(hours).pad_zeros(2) + ":" + str(minutes).pad_zeros(2)
 
-func show_popup(popup_type: String, popup_message: String, accept_function: Callable, decline_function: Callable) -> void:
+func show_popup(_popup_type: String, popup_message: String, accept_function: Callable, decline_function: Callable) -> void:
```
Comment (conalli) [inline scripts/ui/GUI.gd:49]: was generating unused variable warning, this function should probably be removed entirely in separate refactor

File: scripts/scene/RoomBuilder.gd (hunk 4 above)
Comment (khatakslied) [reply scripts/scene/RoomBuilder.gd:35]: No, it's a fair change! Thanks.

File: scripts/scene/RoomBuilder.gd (hunk 6 above)
Comment (khatakslied) [reply scripts/scene/RoomBuilder.gd:228]: Thanks. The old function doesn't make sense anymore. `rooms` should be an array of Room objects. I think this is a hangover from a simpler setup when we were just storing the ids, and I forgot to update it.

File: scripts/ui/GUI.gd (hunk 7 above)
Comment (khatakslied) [reply scripts/ui/GUI.gd:49]: Agreed.

**Review by khatakslied — APPROVED** [review]

**Pull request merged**